File: src/main/java/ru/develgame/jsfgame/CreatePersonBean.java
package ru.develgame.jsfgame;

import ru.develgame.jsfgame.domain.Person;
import ru.develgame.jsfgame.user.UserBean;

import javax.enterprise.context.SessionScoped;
import javax.inject.Inject;
import javax.inject.Named;
import java.io.Serializable;

@Named("createPerson")
@SessionScoped
public class CreatePersonBean implements Serializable {
    @Inject
    private UserBean userBean;

    private static final long serialVersionUID = 9138108733190235780L;

    public Person getPerson() {
        return userBean.getPerson();
    }

    public String game() {
        return "GAME";
    }

    public String chatRoom() {
        return "CHAT";
    }
}

File: src/main/kotlin/ru/develgame/jsfgame/CreatePersonBean.kt
package ru.develgame.jsfgame

import ru.develgame.jsfgame.player.PlayerBean
import java.io.Serializable
import javax.ejb.EJB
import javax.enterprise.context.SessionScoped
import javax.inject.Inject
import javax.inject.Named

@Named("createPerson")
@SessionScoped
open class CreatePersonBean : Serializable {
    @Inject
    private lateinit var playerBean: PlayerBean

    @EJB
    private lateinit var chatBean: ChatBean

    open fun getPerson() = playerBean.getPerson()

    open fun createPerson(): String {
        if (playerBean.getPerson().name.isNullOrEmpty())
            playerBean.getPerson().name = playerBean.getPerson().uuid

        chatBean.personJoined(playerBean.getPerson())

        return "GAME"
    }
}